### PR TITLE
RES-3947: Add option to set TTL on circular buffer

### PR DESCRIPTION
**Pull request by andrewcchau**

### CODE REVIEW

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java [hunk 1]
```diff
@@ -300,4 +300,23 @@ public void peekAllFromEmpty() {
         circularBuffer.clear();
         assertTrue(circularBuffer.isEmpty());
     }
+
+    @Test
+    public void circularBufferTTL() throws InterruptedException {
+        final int ttlInSeconds = 5;
```
Comment (gjesse) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java:306]: 5 seconds is a long time to sleep in a test, let's make this like 1 second

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java [hunk 2]
```diff
@@ -300,4 +300,23 @@ public void peekAllFromEmpty() {
...
+        assertTrue(buffer.isEmpty());
+
+        final UUID first = UUID.randomUUID();
+        final UUID second = UUID.randomUUID();
+        buffer.addAll(ImmutableList.of(first, second));
+        assertEquals(buffer.size(), 2);
+        assertEquals(buffer.peek(), first);
+        assertTrue(buffer.containsAll(ImmutableList.of(first, second)));
+
+        Thread.sleep(ttlInSeconds * 1000);
```
Comment (gjesse) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java:319]: either sleep a bit longer to give a little wiggle room here or use awaitility to give it a little bit of buffer. timing based tests are generally pretty finicky.

File: rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java [hunk 3]
```diff
@@ -300,4 +302,23 @@ public void peekAllFromEmpty() {
...
+        await().atMost(2, TimeUnit.SECONDS).until(buffer::isEmpty);
```
Comment (gjesse) [inline rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java:322]: cool but now you don't need the sleep() :)

Comment (andrewcchau) [reply rdbi-recipes/src/test/java/com/lithium/dbi/rdbi/recipes/queue/RedisCircularBufferTest.java:322]: Whoops

**Pull request merged**